added perf test (#340)
threefoldtech/info_grid: src/internals/zos/performance/performance.md

<h1> Performance Monitor Package </h1>

<h2>Table of Contents</h2>

- [Overview](#overview)
- [Flow](#flow)
- [Node Initialization Check](#node-initialization-check)
- [Scheduling](#scheduling)
- [RMB Commands](#rmb-commands)
- [Caching](#caching)
- [Registered Tests](#registered-tests)
- [Test Suite](#test-suite)

***

## Overview

The `perf` package is a performance monitor in `zos` nodes. it schedules tasks, cache their results and allows retrieval of these results through `RMB` calls.

## Flow

1. The `perf` monitor is started by the `noded` service in zos.
2. Tasks are registered with a schedule in the new monitor.
3. A bus handler is opened to allow result retrieval.

## Node Initialization Check

To ensure that the node always has a test result available, a check is performed on node startup for all the registered tasks, if a task doesn't have any stored result, it will run immediately without waiting for the next scheduled time.

## Scheduling

- Tasks are scheduled using a 6 fields cron format. this format provides flexibility to define time, allowing running tasks periodically or at specific time.

> - Each task have a jitter which is the max number of seconds the task could sleep before it runs, this happens to prevent all tests ending up running at exactly the same time. so for example if a task scheduled to run at `06:00` and it is jitter is `10` it is expected to run any when between `06:00` to `06:10`

Reviewer: ```suggestion
- Each task has a jitter which is the maximum number of seconds the task could sleep before it runs, this happens to prevent all tests ending up running at exactly the same time. so for example if a task is scheduled to run at `06:00` and it is jitter is `10` it is expected to run any when between `06:00` to `06:10`
```
